Microsoft.Research/Contracts/System.Core.Contracts/System.Linq.Expressions.IndexExpression.cs

```diff
@@ -49,18 +50,21 @@ protected internal override Expression Accept (ExpressionVisitor visitor)
       return default(Expression);
     }
 
+    [Pure]
     public IndexExpression Update (Expression @object, IEnumerable<Expression> arguments)
     {
+      Contract.Ensures(Contract.Result<IndexExpression>() != null);
```

Comment:
:thought_balloon: Interesting. The code backs this up, though I've never seen a static indexer (and don't think you can create or use them in C#).